src/extensions/default/bramble/lib/RemoteCommandHandler.js

```diff
@@ -195,6 +196,11 @@ define(function (require, exports, module) {
             skipCallback = true;
             CommandManager.execute("bramble.addCodeSnippet", args[0]).always(callback);
             break;
+        case "INITIALIZE_COLLABORATION":
+            var query = (new URL(window.location.href)).searchParams;
```

Comment:
None of this should be happening here!

You should be passing a config object to the `collaboration` property of `Bramble.load` and have the logic for this happen on the iframe parent instead.

hosted.js
```js
Bramble.load("#bramble", {
    // other options
    collaboration: {
        room: (new URL(window.location.href)).searchParams.get("collaboration")
    }
});
```

main.js
```js
if(options.collaboration) {
    self._executeRemoteCommand({
        commandCategory: "bramble",
        command: "INITIALIZE_COLLABORATION",
        args: options.collaboration
    });
}
```

RemoteCommandHandler.js
```js
case "INITIALIZE_COLLABORATION":
    Collaboration.connect(args[0]);
    break;
```